### ISSUE TITLE
`comparison-overlap` triggers for string enum members when the member name isn't the same as the comparison value

### ISSUE THREAD

**Issue by bzoracler** [python/mypy#19576]
**Bug Report, To Reproduce, and Actual Behaviour**

If the enum member's name to compare isn't equal to the comparing value, a false positive `comparison-overlap` is triggered. See [mypy Playground](https://mypy-play.net/?mypy=master&python=3.12&flags=strict%2Cstrict-equality&gist=78dd3100e5233a27f632d07da7b7fc49):

```python
from enum import Enum

class A(str, Enum):  # Same behaviour if `class A(enum.StrEnum)`
    a = "b"

A.a == "a"  # OK
A.a == "b"  # E: Non-overlapping equality check (left operand type: "Literal[A.a]", right operand type: "Literal['b']")  [comparison-overlap]
```

**Expected Behavior**

No errors

**Your Environment**

- Mypy version used: 1.17, master
- Python version used: 3.12

**Comment by florianvazelle** [comment]
Hi @bzoracler, for me, this is a true positive, because you are comparing an enum value and a string. IMO, the right thing to do is to compare `A.a.value == "b"`, no ? 🤔

**Comment by bzoracler** [comment]
@florianvazelle No, as mentioned in the issue title, this is about *string enums*. String enums (including `enum.StrEnum`) don't need to access `.value` to perform `str` operations or be treated as a `str`, as [explicitly stated in the Python docs](https://docs.python.org/3/library/enum.html#enum.StrEnum):

> _StrEnum_ is the same as [Enum](https://docs.python.org/3/library/enum.html#enum.Enum), but its members are also strings and can be used in most of the same places that a string can be used.

You can try it yourself:

```python
from enum import Enum

class MyStrEnum(str, Enum):
    member = "asdfjkl"
```
```pycon
>>> print(MyStrEnum.member == "asdfjkl")
True
```

**Comment by florianvazelle** [comment]
Ok, thank you for clarifying that 🙏
